### ISSUE TITLE
bridge does not tell clients about network-trouble

### ISSUE THREAD

**Issue by Merovius** [robustirc/bridge#10]
This is potentially an outdated issue, as I used a very old version of the bridge. Other people had similar issues, though.

Around Midnight, for about an hour, my bridge couldn't connect to the network (excerp from log below. I don't have a full log, sorry). This in and off itself is worth a bug, but the main issue with that is, that the clients didn't know that the connection failed, which meant that everything seemed okay to the users. The bridge should really relay such error messages to the client, at least if they persist or something.

```
2015/04/27 23:53:35 sendRequest("/robustirc/v1/0x13d8f89b074b2955/message") failed: Post https://alp.robustirc.net:60667/robustirc/v1/0x13d8f89b074b2955/message: dial tcp 46.20.246.99:60667: connection refused
2015/04/27 23:57:17 sendRequest("/robustirc/v1/0x13d8f89b074b2955/messages?lastseen=1430174064921011690.1") failed: Get https://alp.robustirc.net:60667/robustirc/v1/0x13d8f89b074b2955/messages?lastseen=1430174064921011690.1: dial tcp 46.20.246.99:60667: connection refused
2015/04/27 23:57:18 Protocol error on "dock0.robustirc.net:60667": Could not decode response chunk as JSON: unexpected EOF
2015/04/27 23:57:18 sendRequest("/robustirc/v1/0x13d8f89b074b2955/message") failed: Post https://dock0.robustirc.net:60667/robustirc/v1/0x13d8f89b074b2955/message: EOF
2015/04/28 00:27:43 sendRequest("/robustirc/v1/0x13d8f89b074b2955/messages?lastseen=1430174064921011690.1") failed: Get https://ridcully.robustirc.net:60667/robustirc/v1/0x13d8f89b074b2955/messages?lastseen=1430174064921011690.1: dial tcp 78.46.97.235:60667: connection refused
2015/04/28 00:27:43 sendRequest("/robustirc/v1/0x13d8f89b074b2955/message") failed: Post https://ridcully.robustirc.net:60667/robustirc/v1/0x13d8f89b074b2955/message: dial tcp 78.46.97.235:60667: connection refused
2015/04/28 00:31:25 sendRequest("/robustirc/v1/0x13d8f89b074b2955/messages?lastseen=1430174064921011690.1") failed: Get https://alp.robustirc.net:60667/robustirc/v1/0x13d8f89b074b2955/messages?lastseen=1430174064921011690.1: dial tcp 46.20.246.99:60667: connection refused
```

**Comment by joker234** [comment]
if useful; here is my log since 00:30. Everything before looks like the first lines of this log extract.

```
2015/04/28 00:30:26 Warning: https://ridcully.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/message: dial tcp 78.46.97.235:60667: connection refused (trying different server)
2015/04/28 00:30:39 Warning: https://ridcully.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/messages?lastseen=1430173982527827138.3: Get https://ridcully.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/messages?lastseen=1430173982527827138.3: dial tcp 78.46.97.235:60667: connection refused (trying different server)
2015/04/28 00:30:39 Warning: https://dock0.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/message: dial tcp 79.140.39.208:60667: connection refused (trying different server)
2015/04/28 00:30:39 Warning: https://dock0.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/messages?lastseen=1430173982527827138.3: Get https://dock0.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/messages?lastseen=1430173982527827138.3: dial tcp 79.140.39.208:60667: connection refused (trying different server)
2015/04/28 00:30:58 Warning: https://alp.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/message: unexpected EOF (trying different server)
2015/04/28 00:30:58 Protocol error on "alp.robustirc.net:60667": Could not decode response chunk as JSON: unexpected EOF
2015/04/28 00:32:03 Protocol error on "ridcully.robustirc.net:60667": Could not decode response chunk as JSON: unexpected EOF
2015/04/28 00:32:03 Warning: https://ridcully.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/message: unexpected EOF (trying different server)
2015/04/28 00:32:03 Warning: https://dock0.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/message: 500 Internal Server Error: "No leader known. Please try another server.\n" (trying different server)
2015/04/28 00:33:39 Protocol error on "alp.robustirc.net:60667": Could not decode response chunk as JSON: unexpected EOF
2015/04/28 00:33:39 Warning: https://alp.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/message: unexpected EOF (trying different server)
2015/04/28 00:33:39 Warning: https://dock0.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/message: dial tcp 79.140.39.208:60667: connection refused (trying different server)
2015/04/28 00:33:57 Warning: https://ridcully.robustirc.net:60667/robustirc/v1/0x13d8f96eded93d15/message: 500 Internal Server Error: "No leader known. Please try another server.\n" (trying different server)
```

**Comment by stapelberg** [comment]
I think this is fixed with commit c6df37df038a43b9f4c2580dbf6dfd681e0cbdcb, which eliminates a lockup. With that, I think the following description actually describes reality:

The bridge sends a PING message to the RobustIRC network every minute. If it doesn’t receive a PONG answer to that within another minute, it will kill the session, i.e. disconnect the IRC client. We can’t make it any more explicit than that :).
